docs/developer-guide/using-lighting.md

```diff
@@ -0,0 +1,139 @@
+# Guide to Using Lighting Effect
+
+A deck.gl lighting effect is a visual approximation of environment illumination based on simplified models to make rendering appear more realistic.
+
+To enable lighting in deck.gl, it is required that both the lighting effect and material instances are properly instantiated.
+
+## Constructing Lighting Effect Instance
+
+A [LightingEffect](/docs/effects/lighting-effect.md) can be instantiated with a `lights` object:
+
+```js
+import {AmbientLight, PointLight, DirectionalLight, LightingEffect} from '@deck.gl/core';
+
+// create ambient light source
+const ambientLight = new AmbientLight({
+  color: [255, 255, 255],
+  intensity: 1.0
+});
+// create point light source
+const pointLight = new PointLight({
+  color: [255, 255, 255],
+  intensity: 2.0,
+  // use coordinate system as the same as view state
+  position: [-125, 50.5, 5000]
+});
+// create directional light source
+const directionalLight = new DirectionalLight({
+  color: [255, 255, 255],
+  intensity: 1.0,
+  direction: [-3, -9, -1]
+});
+// create lighting effect with light sources
+const lightingEffect = new LightingEffect({ambientLight, pointLight, directionalLight});
+```
+
+The `lights` has all the light sources that the lighting effect uses to build the visualization. Users typically specify the following types of light sources:
+
+* [AmbientLight](/docs/api-reference/lights/ambient-light.md)
+* [PointLight](/docs/api-reference/lights/point-light.md)
+* [DirectionalLight](/docs/api-reference/lights/directional-light.md)
+
+
+## Constructing Material Instance
+
+A material represents a lighting model specified per layer, a [PhongMaterial](http://uber.github.io/luma.gl/#/documentation/api-reference/phong-material)
+can be instantiated with model parameters.
+
+```js
+import {PhongMaterial} from '@luma.gl/core';
+
+const material =  new PhongMaterial({
+  ambient: 0.2,
+  diffuse: 0.5,
+  shininess: 32,
+  specularColor: [255, 255, 255]
+});
+```
+
+
+## Using Material Instance
+
+To enable lighting for a layer, material prop is required.
```

Comment:
This is not true: all 3D layers have default values for `material. If one wants to disable lighting, they need to manually set `material` to `null`.